src/Frame.cpp

```diff
@@ -480,6 +480,11 @@ const unsigned char* Frame::GetPixels()
 // Get pixel data (for only a single scan-line)
 const unsigned char* Frame::GetPixels(int row)
 {
+	// Check for blank image
```

Comment:
I know... more magic stuff happening in the Frame class :grimacing:

Reply:
Not just magic stuff, _wrong_ stuff. Getters shouldn't be creating data! Heck, that should be a `const` method.

If there's no image in the Frame and something requests pixel data, isn't the right thing to either throw an exception or return `nullptr`? Why does the class do so much work to make it **impossible** to keep image data out of it? I mean, heck, why is there even a `has_image_data` info member, when it's never checked, and it's virtually **impossible** for it to remain `false` even if you TRY?

I feel like all this magical image-creation is trying to make the API more "friendly" by loading it with assumptions about what the caller wants, which is only even maybe-right when they're **correct** assumptions. In fact, even the idea that auto-creating data out of thin air makes the API more friendly is _itself_ an assumption.

If nothing else the comment is wrong. The image will be filled with whatever the value of `color` is, not necessarily black. (And `color` will be set to that argument, at the same time. Which makes me really nervous, now that I look at it.)